### DISCUSSION TITLE
simple selector for export should make a forwarder for its setter

### DISCUSSION THREAD

**Discussion by BarkingBad** [scala/scala3#17903] [Feature Requests]
## Minimized code

```scala
class A:
  var varInt: Int = 1

class B:
  val a = new A
  export a.varInt
  varInt = 2 // Illegal (compile will complain: Reassignment to val varInt)
```

## However

```scala
class A:
  var varInt: Int = 1

class B:
  val a = new A
  export a._
  varInt = 2 // completely fine
```

**Comment by bishabosha** [comment]
The problem with exporting the setter automatically is now if you define an overload of `varInt_=`, and want to also export `varInt_=` overloads, such as below, what should happen?:
```scala
class A:
  var varInt: Int = 1
  def varInt_= : Unit = println("ok") // unrelated overload of setter for `varInt`

class B:
  val a = new A
  export a.varInt // exports `varInt`, `varInt_=(x$1: Int)`
  export a.varInt_=  // exports `varInt_=(x$1: Int)`, `varInt_=`, causing a double definition
  varInt = 2
```

**Comment by bishabosha** [comment]
I will move this to feature requests because the present behaviour is as specified.

deferring to section 4.2 of the Scala language specification, a `var` is equivalent to two definitions, a getter and a setter, which have different names:
https://scala-lang.org/files/archive/spec/2.13/04-basic-declarations-and-definitions.html#variable-declarations-and-definitions

Additionally, as specified at http://dotty.epfl.ch/docs/reference/other-new-features/export.html a simple selector, such as in `export a.varInt` will only export eligable members _with the same name_, i.e. the getter for `varInt` in this case

**Comment by bishabosha** [comment]
This feature request would bring `export` of a `var` to parity for how `import` of a `var` is handled, which also brings setters into scope
